internal/controllers/extension_controller.go

```diff
@@ -123,22 +147,69 @@ func (r *ExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alpha1.Ext
 		return ctrl.Result{}, nil
 	}
 
-	// TODO: kapp-controller integration
-	// * gather variables for resolution
-	// * perform resolution
-	// * lookup the bundle in the solution/selection that corresponds to the Extension's desired Source
-	// * set the status of the Extension based on the respective deployed application status conditions.
+	if !r.HasKappApis {
+		setInstalledStatusConditionFailed(&ext.Status.Conditions, errkappAPIUnavailable.Error(), ext.GetGeneration())
+
+		ext.Status.ResolvedBundleResource = ""
+		setResolvedStatusConditionUnknown(&ext.Status.Conditions, "kapp apis are unavailable", ext.GetGeneration())
+
+		setDeprecationStatusesUnknown(&ext.Status.Conditions, "kapp apis are unavailable", ext.GetGeneration())
+		return ctrl.Result{}, errkappAPIUnavailable
+	}
+
+	// TODO: Improve the resolution logic.
+	bundle, err := r.resolve(ctx, *ext)
+	if err != nil {
+		ext.Status.InstalledBundleResource = ""
+		setInstalledStatusConditionUnknown(&ext.Status.Conditions, "installation has not been attempted as resolution failed", ext.GetGeneration())
+		ext.Status.ResolvedBundleResource = ""
+		setResolvedStatusConditionFailed(&ext.Status.Conditions, err.Error(), ext.GetGeneration())
+		setDeprecationStatusesUnknown(&ext.Status.Conditions, "deprecation checks have not been attempted as resolution failed", ext.GetGeneration())
```

Comment:
Might not be possible based on the way bundle provider works, but in theory we should be able to determine package and channel deprecation status even if bundle selection fails.

I think #599 helps address this, though I haven't reviewed that yet.

Reply:
It would be good to capture this as a non-blocking follow-up improvement.